src/CalcViewModel/GraphingCalculator/EquationViewModel.cpp

```diff
@@ -1,14 +1,316 @@
+                periodicityItem->IsText = false;
+            }
+            break;
+        case 2:
+            periodicityItem->DisplayItems->Append(m_resourceLoader->GetString(L"KGFPeriodicityNotPeriodic"));
+            periodicityItem->IsText = false;
+            break;
+        default:
+            periodicityItem->DisplayItems->Append(m_resourceLoader->GetString(L"KGFPeriodicityError"));
+            periodicityItem->IsText = true;
+        }
+
+        KeyGraphFeaturesItems->Append(periodicityItem);
     }
+
+    void EquationViewModel::SetMonotoncityStringProperty()
+    {
+        KeyGraphFeaturesItem ^ monotonicityItem = ref new KeyGraphFeaturesItem();
+        monotonicityItem->Title = m_resourceLoader->GetString(L"Monotonicity");
+        if (Monotonicity->Size != 0)
+        {
+            for (auto item : Monotonicity)
+            {
+                GridDisplayItems ^ gridItem = ref new GridDisplayItems();
+                gridItem->Expression = item->Key;
+
+                auto monotonicityType = item->Value->Data();
+                switch (*monotonicityType)
+                {
+                case '0':
+                    gridItem->Direction = m_resourceLoader->GetString(L"KGFMonotonicityUnknown");
+                    break;
+                case '1':
+                    gridItem->Direction = m_resourceLoader->GetString(L"KGFMonotonicityIncreasing");
+                    break;
+                case '2':
+                    gridItem->Direction = m_resourceLoader->GetString(L"KGFMonotonicityDecreasing");
+                    break;
+                case '3':
+                    gridItem->Direction = m_resourceLoader->GetString(L"KGFMonotonicityConstant");
+                    break;
+                default:
+                    gridItem->Direction = m_resourceLoader->GetString(L"KGFMonotonicityError");
+                    break;
+                }
+
+                monotonicityItem->GridItems->Append(gridItem);
+            }
+            monotonicityItem->IsText = false;
+        }
+        else
+        {
+            monotonicityItem->DisplayItems->Append(m_resourceLoader->GetString(L"KGFMonotonicityError"));
+            monotonicityItem->IsText = true;
+        }
+
+        KeyGraphFeaturesItems->Append(monotonicityItem);
+    }
+
+    void EquationViewModel::SetTooComplexFeaturesErrorProperty()
+    {
+        if (TooComplexFeatures <= 0)
+        {
+            return;
+        }
+
+        Platform::String ^ separator = ref new String(LocalizationSettings::GetInstance().GetListSeparator().c_str());
+
+        wstring error;
+        if ((TooComplexFeatures & KeyGraphFeaturesFlag::Domain) == KeyGraphFeaturesFlag::Domain)
+        {
+            error.append((m_resourceLoader->GetString(L"Domain") + separator + L" ")->Data());
+        }
+        else if ((TooComplexFeatures & KeyGraphFeaturesFlag::Range) == KeyGraphFeaturesFlag::Range)
```

Comment:
Shouldn't all of these be just if statements instead of else if? Otherwise you only ever capture one feature that is too complex. Also out of curiosity what type of equation is too complex? I only ever get the unsupported error message.

Reply:
good catch.